crates/miden-lib/src/utils/protocol_assembler.rs

```diff
@@ -231,32 +236,75 @@ impl ScriptBuilder {
         mut self,
         module_path: impl AsRef<str>,
         module_code: impl AsRef<str>,
-    ) -> Result<Self, ScriptBuilderError> {
+    ) -> Result<Self, ProtocolAssemblerError> {
         self.link_module(module_path, module_code)?;
         Ok(self)
     }
 
     // SCRIPT COMPILATION
     // --------------------------------------------------------------------------------------------
 
+    /// Compiles an [`AccountComponentCode`] with the provided module path and MASM code.
+    /// The compiled code can be used to create account components.
+    ///
+    /// # Arguments
+    /// * `component_path` - The path to the account code module (e.g., `my_account::my_module`)
+    /// * `component_code` - The account component source code
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - The transaction script compilation fails
+    /// - If `component_path` is not a valid [`LibraryPath`]
+    pub fn compile_component_code(
+        self,
+        component_path: impl AsRef<str>,
+        component_code: impl AsRef<str>,
```

Comment:
For API consistency, I suppose it would be nice if this would take `impl Parse`.

I think we could support this with a little more code:

```rust
let module = component_code
    .parse_with_options(
        &self.source_manager,
        ParseOptions::new(miden_objects::assembly::ModuleKind::Library, lib_path)
            .expect("TODO"),
    )
    .expect("TODO");

let library = assembler
    .assemble_library([module])
    .map_err(|err| {
        ProtocolAssemblerError::build_error_with_report(
            "failed to compile component code",
            err,
        )
    })?;
```